subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -2138,6 +2172,79 @@ static int init_reset(void)
 	return 0;
 }
 
+#if defined(CONFIG_BT_LL_SW_LLCP_LEGACY)
+static void rx_hold_put(struct ll_conn *conn, memq_link_t *link,
+			struct node_rx_pdu *rx)
+{
+#if defined(CONFIG_BT_CTLR_RX_ENQUEUE_HOLD)
+	struct node_rx_pdu *rx_last;
+	struct lll_conn *lll;
+
+	link->mem = NULL;
+	rx->hdr.link = link;
+
+	rx_last = conn->llcp_rx_hold;
+	while (rx_last && rx_last->hdr.link && rx_last->hdr.link->mem) {
+		rx_last = rx_last->hdr.link->mem;
```

Comment:
`rx_last->hdr.link->mem` is always assigned the supplied `rx` which is an allocation by the caller of this function.

Once in the `conn->llcp_rx_hold` linked list, only call to `rx_hold_flush()` would lead to the eventual release of `rx` back to pool. Value in each traversal of `rx_last->hdr.link->mem` will be unique reference to an allocated `struct node_rx_pdu *` pointer.

There will not be an infinite loop, unless an external corruption of memory.

Reply:
> There will not be an infinite loop, unless an external corruption of memory.

So there's no risk of a code bug may introduce an infinite loop, which are notoriously hard to find and debug?

Just wondering if we should insert an __ASSERT that verified that `rx_last != rx_last->hdr.link->mem` before updating the pointer.

Reply:
A simple check cannot detect an external code corrupting a linked. I do not think it is feasible to detect loops in linked lists. Connecting a debugger to a hung board or watchdog should catch infinite loops.

Reply:
Alright, up to you :)